lua/drex/fs.lua

```diff
@@ -245,18 +245,31 @@ function M.scan_directory(path, root_path)
         local name, type = luv.fs_scandir_next(data)
         if not name then break end
 
+        table.insert(content, {name, type})
+    end
+
+    table.sort(content, function (a, b)
+        local aname, atype = a[1], a[2]
+        local bname, btype = b[1], b[2]
+
+        if atype == btype then
+            return aname < bname
+        else
+            return atype < btype
```

Comment:
In most cases this will work fine, but this will group "everything" by type (`directory`, `fifo`, `file`, `link`, `socket`, etc.) instead of just putting the directories first. I guess it might make sense to check for `directory` specifically and let the "rest" be sorted ascending (this is what most OS file manager do)

Reply:
Good idea, implemented